test/unit/domain/thirdpartyRequests/transactions.test.ts

```diff
@@ -0,0 +1,197 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Sridhar Voruganti <sridhar.voruganti@modusbox.com>
+
+ --------------
+ ******/
+
+import { ResponseObject, ResponseToolkit } from "@hapi/hapi"
+import { Transactions } from '~/domain/thirdpartyRequests'
+import Logger from '@mojaloop/central-services-logger'
+import { Util, Enum } from '@mojaloop/central-services-shared'
+import * as ErrorHandler from '@mojaloop/central-services-error-handling'
+import MockData from 'test/unit/data/mockData.json'
+
+const mock_getEndpoint = jest.spyOn(Util.Endpoints, 'getEndpoint')
+const mock_sendRequest = jest.spyOn(Util.Request, 'sendRequest')
+const mock_loggerPush = jest.spyOn(Logger, 'push')
+const mock_loggerError = jest.spyOn(Logger, 'error')
+const api_path = '/thirdpartyRequests/transactions'
+const mock_data = JSON.parse(JSON.stringify(MockData))
+let request = mock_data.transactionRequest
+
+/**
+ * Mock Span
+ */
+class Span {
+  isFinished: boolean
+  constructor() {
+    this.isFinished = false
+  }
+  getChild() {
+    return new Span()
+  }
+  audit() {
+    return jest.fn()
+  }
+  error() {
+    return jest.fn()
+  }
+  finish() {
+    return jest.fn()
+  }
+}
+let MockSpan = new Span()
+
+// @ts-ignore
+const h: ResponseToolkit = {
+  response: (): ResponseObject => {
+    return {
+      code: (num: number): ResponseObject => {
+        return num as unknown as ResponseObject
```

Comment:
this is invalid mocking, ResponseObject is an object and has `statusCode` property to keep status, not a number itself -> so your tests could be already false positive

Reply:
I don't see where you are using `h` in the tests below - maybe I am blind...

Reply:
not using that, deleted unused variable